include/h2o.h

```diff
@@ -1053,23 +1051,27 @@ struct st_h2o_req_t {
      */
     size_t bytes_sent;
     /**
-     * counts the number of times the request has been reprocessed (excluding delegation)
+     * the number of times the request can be reprocessed (excluding delegation)
      */
-    unsigned num_reprocessed;
+    unsigned remaining_reprocesses;
     /**
-     * counts the number of times the request has been delegated
+     * the number of times the request can be delegated
      */
-    unsigned num_delegated;
+    unsigned remaining_delegations;
 
     /**
      * environment variables
      */
     h2o_iovec_vector_t env;
 
     /**
-     * error logs
+     * error logging
      */
-    H2O_VECTOR(h2o_req_error_log_t) error_logs;
+    struct {
+        h2o_buffer_t *buf;
+        void (*cb)(h2o_req_t *req, void *data, h2o_iovec_t error);
+        void *data;
+    } error;
```

Comment:
Would you mind renaming this to `error_log` or `error_logger`? IMO this is an object that logs the error, not the error itself.